heapsort

ugin331 opened the pull request:
no comment

davidxl commented on sort-animation/public_html/heapsort.js:
```diff
@@ -0,0 +1,59 @@
+
+var array = [1000, 4000, 6000, 3000, 2000, 5000, 7000, 10000, 8000, 9000];
+var newarray = [];
+function leftroot(i){
```
leftroot --> left_child

davidxl commented on sort-animation/public_html/heapsort.js:
```diff
@@ -0,0 +1,59 @@
+function rightroot(i){
+    var reeto = 2*i+2;
+    return reeto;
+}
```
right_child

davidxl commented on sort-animation/public_html/heapsort.js:
```diff
@@ -0,0 +1,59 @@
+    var reeto = 2*i+2;
+    return reeto;
+}
+function detectleftroot(arraypos){
```
add a comment for this function.

davidxl commented on sort-animation/public_html/heapsort.js:
```diff
@@ -0,0 +1,59 @@
+}
+function heapify(rootnum){
+    console.log(rootnum);
+    while(detectleftroot===true){
```
comparing function (not function call) with true?

davidxl commented on sort-animation/public_html/heapsort.js:
```diff
@@ -0,0 +1,59 @@
+    array[swap1] = nexvar; //sets the original array spot with the swapped one
+    console.log(array);
+}
+function heapify(rootnum){
```
you are mixing up heapify with sorting. 

1) Heapify does not need to do sifting.
2) Heapify is recursive
3) Heapify does not need a while loop
4) Heapify is called by the sorting method once at the root
4) Siftdown will be called many times (in a loop) by the sorter function.

ugin331 commented:
So I changed the thing around a bit but I don't quite understand. The heaping function doesn't recurse? The heaping function is the one making the heaps tho, wouldn't you recursively call heap on the smaller pieces then sifting the small bits down? I changed heap to sort, but what does that do
